### PR TITLE
Cccp set

### PR DESCRIPTION
**Pull request by VinInn**
Factorize CCC value into a single cfi file
Factories the reading of this PSet into a single header file
use consistently.

No regression expected, no regression observed. HLT expected to fail.
@mtosi @perrotta please check/act/instruct

affected producers

SiStripClusterizer, SiStripClusterizerFromRaw (actually their  Algorithm''s PSet)
SeedingLayersEDProducer (actually its embedded strip PSets)
Chi2ChargeMeasurementEstimatorESProducer
ClusterChargeMasker
ClusterShapeHitFilterESProducer

in all cases the optional attribute
"minGoodStripCharge"/"minGoodCharge"
has been substituted with a MANDATORY

```
clusterChargeCut = cms.PSet(refToPSet_ = cms.string('SiStripClusterChargeCutXXX'))
```

with XXX = None/Loose/Tight
corresponding to current CCC values of 0/1724/2069

Lesson Learned:
use of ParameterSet::existsAs is nothing less that introducing a time bomb that will at some point induce subtle regressions such as massive HLT/ALCA silent failures...

### CODE REVIEW

**Comment by Dr15Jones** [comment]
I concur with your "lessons learned".

File: FastSimulation/Tracking/python/IterativePixelLessStep_cff.py [hunk 1]
```diff
@@ -33,6 +33,8 @@
 #                                     'TEC1_pos+TEC2_pos','TEC2_pos+TEC3_pos','TEC3_pos+TEC4_pos','TEC3_pos+TEC5_pos','TEC4_pos+TEC5_pos',
 #                                     'TID1_neg+TID2_neg','TID2_neg+TID3_neg',
 #                                     'TEC1_neg+TEC2_neg','TEC2_neg+TEC3_neg','TEC3_neg+TEC4_neg','TEC3_neg+TEC5_neg','TEC4_neg+TEC5_neg']
+
+from RecoLocalTracker.SiStripClusterizer.SiStripClusterChargeCut_cfi import *
```
Comment (lveldere) [inline FastSimulation/Tracking/python/IterativePixelLessStep_cff.py]: @VinInn 
How does this affect FastSim?
FastSim has no real charge.
Lukas

**Comment by Martin-Grunewald** [comment]
Hi @VinInn 
The statement ```HLT expected to fail.``` is kind of useless!
Please do provide instructions on what do change in HLT!

File: FastSimulation/Tracking/python/IterativePixelLessStep_cff.py (hunk 1 above)
Comment (VinInn) [inline FastSimulation/Tracking/python/IterativePixelLessStep_cff.py]: in the following line you import pixelLessStepSeedLayers that makes use of it
There is most probably a different way to import it silently, at the moment my expertise in python (and understanding of RefToPset) makes the current solution the easier

**Comment by venturia** [comment]
I am fine to couple  calZeroBiasClusters to the standard siStripClusters module but someone has to inform strip local reco people and be sure that this information is written in the documentation of the bad channel workflow. Next DPG meeting?

**Comment by VinInn** [comment]
@venturia, I find cleaner, clearer and more maintainable to clone and modify the few (in this case one) parameters that differs w/r/t default than copy/paste/edit

File: FastSimulation/Tracking/python/IterativePixelLessStep_cff.py (hunk 1 above)
Comment (lveldere) [inline FastSimulation/Tracking/python/IterativePixelLessStep_cff.py]: Are you sure this is needed"
We only use the layer list from the seedlayer, 
and copy it to the cfg of the fast trajectory seed producer.
The module itself is not run.

May be there is something I don't understand about the python processing...

**Comment by venturia** [comment]
@VinInn . Agreed, but they have to know that now they depend on something they did not depend on before.

File: FastSimulation/Tracking/python/IterativePixelLessStep_cff.py (hunk 1 above)
Comment (VinInn) [inline FastSimulation/Tracking/python/IterativePixelLessStep_cff.py]: It fails w/o
I acknowledge that there is definitively something "I" don't understand about the python processing of RefToPSet

Comment (makortel) [inline FastSimulation/Tracking/python/IterativePixelLessStep_cff.py]: How does it fail? (I would also have expected this configuration to work without the import above)

Comment (VinInn) [inline FastSimulation/Tracking/python/IterativePixelLessStep_cff.py]:
```----- Begin Fatal Exception 11-Feb-2015 12:40:37 CET-----------------------
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named ZEE_13TeV_TuneCUETP8M1_cfi_GEN_SIM_RECO_EI_HLT_VALIDATION.py
Exception Message:
python encountered the error: <type 'exceptions.AttributeError'>
'Process' object has no attribute 'SiStripClusterChargeCutTight'
```

Comment (makortel) [inline FastSimulation/Tracking/python/IterativePixelLessStep_cff.py]: I think what happens is the following
- from the import below it follows that a `process` loading this configuration will have `process.pixelLessStepSeedLayers` member, that has `refToPSet_`s to `SiStripClusterChargeCutTight`
- all `refToPSet_`s are resolved when the `process` is "transformed" to C++ ParameterSets (`cms.Process.fillProcessDesc()`)
- if the import above is missing, the `process` does not have the `SiStripClusterChargeCutTight` member referenced by a `refToPSet_` in `process.pixelLessStepSeedLayers`

I have one idea to circumvent this "behind the scenes" dependence, but I want to test it first myself.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
it's somewhat pointless to test now, workflows including HLT will fail.
The changes needed for HLT should be provided.

**Comment by Martin-Grunewald** [comment]
@slava77 
This is not planned: I do not know which release this will end up with, 74X or 75X.
I already wasted time on https://github.com/cms-sw/cmssw/pull/7589 which
David Lange told me will after all not go in 74X but rather 75X.
So I do not see the point making any HLT changes unless we know for a fact
that it goes in 74X, and also when!

**Comment by slava77** [comment]
@Martin-Grunewald 
I think this is for 75X.

**Comment by davidlange6** [comment]
But indeed, needing to know the release that a new development goes into is imho the wrong approach. If you need to hack around, at least hack around in the backport as then its clear when the hack can be removed.

**Comment by Martin-Grunewald** [comment]
@davidlange6 
David, you are so long with CMSSW that you should know the constraints
TSG with ConfDB is operating under so refrain from such comments.
If you think TSG does crap have a chat with Andrea B.

**Comment by davidlange6** [comment]
I didn't say anything to suggest this.. rather a better way to live within the constraints imposed by the TSG.

**Comment by davidlange6** [comment]
closing in 74x - there is the same PR open in 75x (now the development release)

**Comment by Martin-Grunewald** [comment]
@VinInn 

Please re-open this PR.

**Comment by VinInn** [comment]
was closed by @davidlange6

**Comment by Martin-Grunewald** [comment]
@VinInn 
While TTRH does not have merge conflicts, this one now has - 
could you please have a look fix?

**Comment by Martin-Grunewald** [comment]
It seems the only merge conflict is in 
```
 RecoLocalTracker/SiStripClusterizer/python/DefaultClusterizer_cff.py
```
where some other parameters have been removed in the meantime.
Could you please update this PR for that (and its 75X twin)?
Thanks

**Comment by VinInn** [comment]
should be ok now

**Comment by slava77** [comment]
-1

superseded by #8343

**Pull request closed without merging**